pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -258,25 +261,6 @@ private String getCommonBrokerTenant(List<String> tableNames) {
     return (String) (tableBrokers.toArray()[0]);
   }
 
-  // return the serverTenant if all tables point to the same broker, else returns null
```

Comment:
Why delete this ?

Reply:
As stated in the description of PR, it's not the right condition to enforce. We could have tables that are hosted in disjoint server tenants. 

> After discussing with the authors, removed the restriction enforced in https://github.com/apache/pinot/pull/10336 where a multistage query should have at least one common server tenant. This restriction might not work when the tables in the query belong to disjoint server tenants.